code/game/machinery/_machinery.dm

```diff
@@ -798,10 +827,14 @@
 	if(SEND_SIGNAL(user, COMSIG_TRY_USE_MACHINE, src) & COMPONENT_CANT_USE_MACHINE_TOOLS)
 		return ITEM_INTERACT_BLOCKING
 
+	//takes priority in case material container or other atoms that hook onto item interaction signals won't give it a chance
+	if(istype(tool, /obj/item/storage/part_replacer))
+		update_last_used(user)
+		return tool.interact_with_atom(src, user, modifiers)
+
 	. = ..()
 	if(.)
 		update_last_used(user)
-	return .
```

Comment:
@MrMelbert would like your thoughts on this since it changes behavior

Reply:
It does? That wasn't the intention so do I put back `return .`

Wait how does it change behavior?

Reply:
Looks fine